src/Veldrid/MTL/MTLFeatureSupport.cs

```diff
@@ -1,17 +1,20 @@
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using Veldrid.MetalBindings;
 
 namespace Veldrid.MTL
 {
-    internal class MTLFeatureSupport
+    public class MTLFeatureSupport : IReadOnlyCollection<MTLFeatureSet>
```

Comment:
I'm not sure about directly exposing this as a public type. Some of the things here will feel a little strange to the average user: they only exist because they are interesting or useful to Veldrid's implementation. I'd actually be more comfortable directly exposing `_supportedFeatureSets` since it's directly from Metal.

Reply:
I actually like the IReadOnlyCollection idea you have with OpenGL extensions. The only useful thing `MTLFeatureSupport` provides though is the `MaxFeatureSet` property, which on second thought can simply be added to `BackendInfoMetal`